alpaca_trade_api/stream2.py

```diff
@@ -271,23 +276,46 @@ def run(self, initial_channels=[]):
         initial_channels is the channels to start with.
         '''
         loop = self.loop
-        try:
-            loop.run_until_complete(self.subscribe(initial_channels))
-            loop.run_until_complete(self.consume())
-        except KeyboardInterrupt:
-            logging.info("Exiting on Interrupt")
-        finally:
-            loop.run_until_complete(self.close())
-            loop.close()
-
-    async def close(self):
-        '''Close any of open connections'''
+        should_renew = True  # should renew connection if it disconnects
+        while should_renew:
+            try:
+                if loop.is_closed():
+                    self.loop = asyncio.new_event_loop()
+                    loop = self.loop
+                loop.run_until_complete(self.subscribe(initial_channels))
+                loop.run_until_complete(self.consume())
+            except KeyboardInterrupt:
+                logging.info("Exiting on Interrupt")
+                should_renew = False
```

Comment:
How effective is this? It won't use this variable in the case of other exception?

Reply:
Oh I see, the it exists the loop.

Reply:
basically we want to reconnect for any exception not handled by the user

Reply:
I would even remove the "except KeyboardInterrupt" because if the user breaks execution everything stops anyway